### ISSUE TITLE
Orleans CodeGenerator (MSBuilld) fails when DTOs are placed in separate namespace

### ISSUE THREAD

**Issue by syberside** [dotnet/orleans#7140]
Assuming following folder/namespace structure
```
Grains.Interfaces.dll - NS Grains.Interfaces
`-Models
  `-MyModel.cs - NS Grains.Interfaces.Models
`-IMyGrain.cs - NS Grains.Interfaces
```

Generated code will be placed in NS Grains.Interfaces, but MyModel references will not contains NS.
In the same time Interface reference will always use full NS in generated code.

**Comment by ReubenBond** [comment]
Hi @syberside, I don't understand why this would happen. Are you able to provide build logs?

**Comment by syberside** [comment]
Hi @ReubenBond, thanks for responding

I’ll create min repo to reproduce it or send logs from my project

**Comment by syberside** [comment]
Hi @ReubenBond 

I tried to reproduce this bug in min repo, but didn't succeed with it.

But also I wound interesting fact: code generation is broken only if *record* type is moved to another namespace.
You can refer to [this ](https://github.com/syberside/tick-tack-toe-orleans-signalr-flutter/compare/reproduce-broken-code-gen) branch and this [commit ](https://github.com/syberside/tick-tack-toe-orleans-signalr-flutter/commit/601ade0667d3c175b37478e84b70c08ebb517ed0#diff-e643e2a662e88df07432688b07a61401e8bf819787024dac617caf750ae9d755R6) for reproduce.

If you will move GameId to Interfaces namespace (remove Dtos) - code generation will start to work as expected.

Also, if you will move class/enum to Dtos (GameAxis, GameMapDto, etc) - code generation will be not broken.

**Comment by ReubenBond** [comment]
This seems like an issue with `record` support, which was fixed in #7108
